tests/unit/test_query_language.py

```diff
@@ -320,6 +323,67 @@ class some_table(EventFrame):
             some_int = Series(int)
 
 
+def test_table_from_file(tmp_path):
+    rows = [(1, 100), (2, 200)]
+    csv_path = tmp_path / "test.csv"
+    with csv_path.open("w") as f:
+        writer = csv.writer(f)
+        writer.writerow(("patient_id", "n"))
+        writer.writerows(rows)
+
+    @table_from_file(csv_path)
+    class some_table(PatientFrame):
+        some_int = Series(int)
+
+    assert isinstance(some_table, PatientFrame)
+    assert isinstance(some_table.qm_node, InlinePatientTable)
+
+
+def test_table_from_file_only_accepts_patient_frame():
+    with pytest.raises(
+        SchemaError,
+        match="`@table_from_file` can only be used with `PatientFrame`",
+    ):
+
+        @table_from_file([])
+        class some_table(EventFrame):
+            some_int = Series(int)
+
+
+def test_table_from_file_only_accepts_known_filetypes(tmp_path):
```

Comment:
Once the file handling logic is moved to into `file_formats/__init__.py` it would make sense to move some of these tests to `test_file_formats.py`.